### PR TITLE
Issue 321: Rewrite syslog to use a queue

### PR DESCRIPTION
**Pull request by AnDann**
Compiles but untested. Just for discussion

### CODE REVIEW

**Comment by AnDann** [comment]
tested and everything seems to work. Need 40byte more flash but saves 408byte ram.
I think it can be merged.

**Comment by eku** [comment]
Goot work. Would you please consolidate the changes to one commit (create new branch and apply patch containing all changes). Thanx.

File: protocols/syslog/syslog.h [hunk 1]
```diff
@@ -27,20 +28,14 @@
 #include <avr/pgmspace.h>
 #include "protocols/uip/uip.h"
 
-#define MAX_DYNAMIC_SYSLOG_BUFFER 500
+#define MAX_DYNAMIC_SYSLOG_BUFFER 100
 
 uint8_t syslog_send_P(PGM_P message);
```
Comment (eku) [inline protocols/syslog/syslog.h]: The API could be slimmed down.

File: protocols/syslog/syslog_net.c [hunk 2]
```diff
@@ -39,7 +39,8 @@ syslog_net_init(void)
 
   syslog_conn = uip_udp_new(&ip, HTONS(SYSLOG_PORT), syslog_net_main);
 
-  if(! syslog_conn) {
+  if (!syslog_conn)
+  {
     debug_printf("syslog: couldn't create connection\n");
```
Comment (eku) [inline protocols/syslog/syslog_net.c]: Ouch, could result in a recursion!

Comment (AnDann) [inline protocols/syslog/syslog_net.c]: syslog_net_init is only called once during startup i thought.

File: protocols/syslog/syslog.h (hunk 1 above)
Comment (AnDann) [inline protocols/syslog/syslog.h]: Yes this is possible but the syslog_sendf functions need more ram because the resulting string length is unkown.

**Comment by AnDann** [comment]
So should i move syslog_queue to core/queue and remove the syslog prefix from the functions?

File: protocols/syslog/syslog_net.c (hunk 2 above)
Comment (eku) [inline protocols/syslog/syslog_net.c]: We are both right. The debug output should be enclosed in #ifndef DEBUG_USE_SYSLOG.

**Comment by AnDann** [comment]
Replaced by PR #401

**Pull request closed without merging**